django_x509/base/models.py

```diff
@@ -429,6 +429,23 @@ def _add_extensions(self, cert):
             ])
         return cert
 
+    def _renew(self):
+        """
+        Renew certificate, private key
+        and validity end date
+        """
+        if hasattr(self, 'ca'):
+            self.ca._generate()
+            self.ca.validity_end = default_ca_validity_end()
+            self.ca.serial_number = uuid.uuid4().int
+            self.ca.save()
+            self.validity_end = default_cert_validity_end()
+        else:
+            self.validity_end = default_ca_validity_end()
+        self._generate()
+        self.serial_number = uuid.uuid4().int
```

Comment:
the instructions to generate a new serial number is already present in the code and is being repeated here, hence it's better to move this to a private method (eg: `_generate_serial_number()`) to ensure future maintainers won't accidentally forget to upgrade part of the code if needed